### PR TITLE
Integration changes and code maintenance

### PR DESCRIPTION
**Pull request by HassanSE19**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +23,7 @@
     "react-router-dom": "^6.27.0",
     "react-scripts": "5.0.1",
     "redux-saga": "^1.3.0",
+    "store": "^2.0.12",
```
Comment (mtalha-mergestack) [inline package.json]: remove this dependency.

File: package.json [hunk 2]
```diff
@@ -50,6 +53,9 @@
     ]
   },
   "devDependencies": {
+    "@types/js-cookie": "^3.0.6",
+    "@types/store": "^2.0.5",
```
Comment (mtalha-mergestack) [inline package.json]: this one too

File: src/containers/LayoutContainer/index.ts [hunk 3]
```diff
@@ -4,8 +4,11 @@ import Layout from "components/Layout";
 import { ILayoutContainerProps } from "type";
 import { USER_ACTION_TYPES } from "utils/constants/actionTypes";
 
-const mapStateToProps = (_: RootState, ownProps: ILayoutContainerProps) => ({
-  withLogout: ownProps.withLogout,
+const mapStateToProps = (
+  state: RootState,
+  ownProps: ILayoutContainerProps
+) => ({
+  isAuthenticated: new Boolean(state.user._id).valueOf(),
```
Comment (mtalha-mergestack) [inline src/containers/LayoutContainer/index.ts]: whats this?

File: src/pages/NotFound/index.tsx [hunk 4]
```diff
@@ -21,13 +21,13 @@ const NotFound = () => {
           </p>
           <button
             className="w-full p-4 md:w-[8rem] bg-[gray] rounded-lg"
-            onClick={() => navigate("/home")}
+            onClick={() => navigate("/", { replace: true })}
```
Comment (HassanSE19) [inline src/pages/NotFound/index.tsx]: use windows.location

**Pull request merged**